core/src/marker/markerManager.cpp

```diff
@@ -14,15 +15,20 @@
 
 namespace Tangram {
 
+MarkerManager::MarkerManager() {}
+
+MarkerManager::~MarkerManager() {}
+
 void MarkerManager::setScene(std::shared_ptr<Scene> scene) {
 
     m_scene = scene;
     m_mapProjection = scene->mapProjection().get();
-    m_styleContext.initFunctions(*scene);
-    m_jsFnIndex = scene->functions().size();
 
+    m_styleContext = std::make_unique<StyleContext>();
+    m_styleContext->initFunctions(*scene);
 
     // Initialize StyleBuilders.
+    m_styleBuilders.clear();
```

Comment:
In accordance with the discussion I had with @karimnaaji for https://github.com/tangrams/tangram-es/pull/1467/files, it make sense to remove all markers on a scene reset.

Reason being, a marker could have been using some style in its styling string or a drawGroup which no longer exists in the new scene. So it becomes the responsibility of the user to re-add/re-style these markers.